docs/internal/DistributedArchitectureGuide.md

```diff
@@ -12,6 +12,214 @@ A guide to the general Elasticsearch components can be found [here](https://gith
 
 # Networking
 
+Every elasticsearch node maintains various networking clients and servers,
+protocols, and synchronous/asynchronous handling.
+
+## HTTP Transport
+
+Every node has an HTTP Transport Server (simply HTTP Transport) and it is a
+single entry point for all external clients (except cross-cluster). Management,
+ingestion, search, and everything else passes through HTTP server.
+
+Elasticsearch works over HTTP 1.1 and supports HTTP features such as: TLS,
+chunked transfer encoding, content compression, pipelining. Although we try to
+be HTTP spec compliant, Elastic is not a webserver. We support GET requests with
+payload (some old proxies might drop content), requests cannot be cached by
+middle boxes.
+
+By default HTTP server binds to the first available port in range
+9200-9300. There is no connections limit, but there is a limit on payload
+size. Default maximum payload is 100mb after compression. It's a very large
+number and almost never a good target that client should approach.
+
+Security is not enabled by default, meaning no TLS and authentication. These
+features are available in the x-pack/security module.
+
+HTTP transport provides two options for content processing: aggregate fully and
+incremental. Aggregated content is a preferable choice for a small messages that
+cannot be parsed incrementally (like JSON). But aggregation has drawbacks, it
```

Comment:
> cannot be parsed incrementally (like JSON)

Mmm sorta depends what you mean by "incrementally" but I'd say that the SAX-style parsing we do is in fact working incrementally. It doesn't really make sense to start parsing before we've received the whole body, but that's different from saying "cannot".